.github/labeler.yml

```diff
@@ -123,3 +123,13 @@
 "area: Modules":
   - "west.yml"
   - "modules/**/*"
+"area: Shields":
+  - "boards/shields/**"
+  - "samples/shields/**"
+"platform: STM32":
+  - "boards/arm/nucleo_*/**"
+  - "boards/arm/*stm32*/**"
+  - "drivers/**/*stm32*"
+  - "dts/arm/st/*/*"
```

Comment:
You could add the driver bindings in the dts folder, also: sth like `dts/**/*stm32*`.